core/src/spotify_id.rs

```diff
@@ -1,6 +1,7 @@
 use byteorder::{BigEndian, ByteOrder};
 use std;
 use std::fmt;
+use std::io::Result;
```

Comment:
It should be using a custom error type, not `io::Error`

Reply:
Something like ```SpotifyIdError``` sound ok?

Reply:
Ok, why should it not use io::Error? Also, I'm unfamiliar with how one implements a custom error type, so any pointers would be much appreciated.

Reply:
Looks like I was tired. Have added the ```SpotifyIdError``` and changed methods to use it accordingly. Please re-review as appropriate.